subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -399,13 +354,11 @@ File collections have some important attributes in Gradle. They can be:
 
 _Lazy creation_ of a file collection is useful when you need to evaluate the files that make up a collection at the time a build runs. In the following example, we query the file system to find out what files exist in a particular directory and then make those into a file collection:
 
-=== Example: Implementing a file collection
-
-[source.multi-language-sample,groovy]
-.build.gradle
-----
-include::{samplesPath}/userguide/files/fileCollections/build.gradle[tag=closure]
-----
+.Implementing a file collection
+====
+include::sample[dir="userguide/files/fileCollections/groovy",files="build.gradle[tags=closure]"]
+include::sample[dir="userguide/files/fileCollections/kotlin",files="build.gradle.kts[tags=closure]"]
+====
```

Comment:
:x: the two paragraphs below this sample are very much Groovy oriented and need to be adjusted

```adoc
[.multi-language-text.lang-groovy]
 The key to lazy creation is passing a closure to the `files()` method. Your closure simply needs to return a value of a type accepted by `files()`, such as `List<File>`, `String`, `FileCollection`, etc.

[.multi-language-text.lang-groovy]
 _Iterating over a file collection_ can be done through the `each()` method on the collection or using the collection in a `for` loop. In both approaches, the file collection is treated as a set of `File` instances, i.e. your iteration variable will be of type `File`.

[.multi-language-text.lang-kotlin]
 The key to lazy creation is passing a `Provider` to the `files()` method. Your provider simply needs to return a value of a type accepted by `files()`, such as `List<File>`, `String`, `FileCollection`, etc.

[.multi-language-text.lang-kotlin]
 _Iterating over a file collection_ can be done through the `forEach()` method on the collection or using the collection in a `for` loop. In both approaches, the file collection is treated as a set of `File` instances, i.e. your iteration variable will be of type `File`.
```